addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -109,11 +109,18 @@ export class DiscussCoreCommon {
                 thread: { id: channel_id, model: "discuss.channel" },
             });
         });
+        this.busService.subscribe("discuss.channel.member/read", (payload) => {
+            const member = this.store.ChannelMember.insert(payload.member);
+            if (member.eq(member.thread.selfMember)) {
+                member.thread.updateUnreadAndNeedactionCounters(member.new_message_separator);
+            }
+        });
         this.busService.subscribe("discuss.channel.member/seen", (payload) => {
             const { channel_id, guest_id, id, last_message_id, partner_id } = payload;
             const member = this.store.ChannelMember.insert({
                 id,
                 seen_message_id: last_message_id ? { id: last_message_id } : null,
+                new_message_separator: last_message_id ? { id: last_message_id } : null,
```

Comment:
seen and separator should be decoupled, seen is for other members, separator should only be send to self